hbase-server/src/main/java/org/apache/hadoop/hbase/master/AssignmentManager.java

```diff
@@ -3343,6 +3346,55 @@ boolean waitUntilNoRegionsInTransition(final long timeout)
     return offlineServers;
   }
 
+  void deleteNonZkBasedQualifiersForZkBasedAssignment() throws IOException {
+    boolean isZKAssignmentInUse =
+      ConfigUtil.useZKForAssignment(server.getConfiguration()) && !server.getConfiguration()
+        .getBoolean("hbase.assignment.usezk.migrating", false);
+    if (isZKAssignmentInUse) {
+      List<Result> results = MetaTableAccessor.fullScanOfMeta(server.getConnection());
+      List<Delete> redundantCQDeletes = new ArrayList<>();
+      for (Result result : results) {
+        RegionLocations rl =  MetaTableAccessor.getRegionLocations(result);
+        if (rl == null) {
+          continue;
```

Comment:
are these `continue` statements ever expected? Do they at least deserve a `LOG` statement at `WARN` or at least `INFO`?

Reply:
I think you want the same log as what happens if locations comes back as null if you managed to get a rl object.

         LOG.error("No location found for " + result);